packages/core/src/utils.js

```diff
@@ -1139,7 +1139,10 @@ export function toPathSchema(schema, name = "", rootSchema, formData = {}) {
     return toPathSchema(_schema, name, rootSchema, formData);
   }
 
-  if (schema.hasOwnProperty("additionalProperties")) {
+  if (
+    schema.hasOwnProperty("additionalProperties") &&
+    schema.additionalProperties === true
+  ) {
```

Comment:
Thanks can you add a test?

Reply:
Test added; I amended the previous commit